### PR TITLE
God renaming and more god types

### PR DESCRIPTION
**Pull request by mrSheploo**
Take a look if everything is ok, please.
Added lines to code:

botutils.py
Lines 79-84 & 133-134

GodManager.py
Lines 171-178

AdminManager.py
Lines 147-154

BotManager.py
Lines 49-51

database.py
Lines 181-184

### CODE REVIEW

**Review by HeroGamers — CHANGES_REQUESTED** [review]
You have copied all of the cogs to the root directory?
Please make changes to the actual files instead of creating duplicate files, as the bot won't run with this structure.

**Comment by mrSheploo** [comment]
I tried to make all files to be in their own directiories. Well like I said
it was my first time using github.

I tried pushing my edits via github desctop app using git push command.
However then it crashes. I'll try doing it again tomorow (10:30pm now here)
in a proper way.

2021-06-22, an 21:13, Hero ***@***.***> rašė:

**Comment by HeroGamers** [comment]
Take your time! Thanks for the work!

**Comment by mrSheploo** [comment]
Hi again!
I hope that I managed to put all the files into their original directories.
Hope it works ;)

File: database.py [hunk 1]
```diff
@@ -1,517 +1,522 @@
...
+    if god.InviteOnly:
+        access = False
+
+    query = gods.update(InviteOnly=access).where(gods.ID == godid)
+    query.execute()
+    return access
+
+# Rename a God Added by mrSheploo
+def rename(godid, godname):
+    query =gods.update(Name=name).where(gods.ID == godid)
```
Comment (HeroGamers) [inline database.py]: No parameter called name, you called it godname.

File: cogs/GodManager.py [hunk 2]
```diff
@@ -168,6 +168,15 @@ async def _setgender(self, ctx, arg1):
             database.setGender(god.ID, arg1)
             await ctx.send("Gender successfully set to: " + arg1 + "!")
 
+    @commands.command(name="rename", aliases=["rn"]) #My attempt to make a possibility to change god's name (mrSheploo)
```
Comment (HeroGamers) [inline cogs/GodManager.py]: Remove your name from this line - people can see that you made it by looking at the commit history.

File: cogs/GodManager.py [hunk 3]
```diff
@@ -168,6 +168,15 @@ async def _setgender(self, ctx, arg1):
...
+    @commands.check(utilchecks.isPriest)
+    async def _rename(self, ctx, arg1):
+        """Rename existing God."""
+        god = database.getBeliever(ctx.authos.id, ctx.guild.id).God
+
```
Comment (HeroGamers) [inline cogs/GodManager.py]: Please also ensure that there is no other God with that name already, by using the getGodName function in the database.
See the create command in GodManager (https://github.com/HeroGamers/DiscordGods/blob/master/cogs/GodManager.py#L30).

File: cogs/AdminManager.py [hunk 4]
```diff
@@ -144,6 +144,16 @@ async def _forcedeletegod(self, ctx, arg1):
         else:
             await ctx.send("A god with that name doesn't exist!")
 
+    @commands.command(name="forcerename", aliases=["frn"]) #My attempt to make a possibility to change god's name (mrSheploo)
+    @commands.has_permissions(administrator=True)
+    async def _rename(self, ctx, arg1):
```
Comment (HeroGamers) [inline cogs/AdminManager.py]: Call it the same name as the name argument on the command (forcerename) for consistency.

File: cogs/AdminManager.py [hunk 5]
```diff
@@ -144,6 +144,16 @@ async def _forcedeletegod(self, ctx, arg1):
...
+        """Rename existing God."""
+        god = database.getGodName(arg1, ctx.guild.id)
+
```
Comment (HeroGamers) [inline cogs/AdminManager.py:164]: Ensure that there is no other God with that name already, by using the getGodName function in the database.
See the create command in GodManager (https://github.com/HeroGamers/DiscordGods/blob/master/cogs/GodManager.py#L30).

File: cogs/AdminManager.py [hunk 6]
```diff
@@ -144,6 +144,16 @@ async def _forcedeletegod(self, ctx, arg1):
         else:
             await ctx.send("A god with that name doesn't exist!")
 
+    @commands.command(name="forcerename", aliases=["frn"]) #My attempt to make a possibility to change god's name (mrSheploo)
```
Comment (HeroGamers) [inline cogs/AdminManager.py]: Remove your name - can be seen in commit history.

File: Util/botutils.py [hunk 7]
```diff
@@ -75,7 +75,13 @@ class botutils:
                 ("PARTY", discord.Color.magenta()),
                 ("WAR", discord.Color.dark_red()),
                 ("WISDOM", discord.Color.dark_purple()),
-                ("NATURE", discord.Color.green())]
+                ("NATURE", discord.Color.green()),
+                ("SACRIFICE", discord.Color.red()), #Added by mrSheploo
```
Comment (HeroGamers) [inline Util/botutils.py]: Remove your name - can be seen in commit history.

File: Util/botutils.py [hunk 8]
```diff
@@ -124,6 +130,8 @@ def getGodString(cls, god):
                 god_title = "Diety"
             elif god.Gender.upper() == "Boeing AH-64 Apache".upper():
                 god_title = "Attack Helicopter"
+            elif god.Gender.upper() == "Machine".upper(): #added by mrSheploo
```
Comment (HeroGamers) [inline Util/botutils.py]: Remove your name - can be seen in commit history.

File: cogs/GodManager.py [hunk 9]
```diff
@@ -168,6 +168,15 @@ async def _setgender(self, ctx, arg1):
             database.setGender(god.ID, arg1)
             await ctx.send("Gender successfully set to: " + arg1 + "!")
 
+    @commands.command(name="rename", aliases=["rn"]) #My attempt to make a possibility to change god's name (mrSheploo)
+    @commands.check(utilchecks.isPriest)
+    async def _rename(self, ctx, arg1):
+        """Rename existing God."""
+        god = database.getBeliever(ctx.authos.id, ctx.guild.id).God
```
Comment (HeroGamers) [inline cogs/GodManager.py]: You accidentally typed authos instead of author (ctx.authos.id -> ctx.author.id).

**Review by HeroGamers — CHANGES_REQUESTED** [review]
I have requested a few changes, but other than that, it looks like a decent first commit!
Good job~

File: cogs/AdminManager.py [hunk 10]
```diff
@@ -144,14 +144,26 @@ async def _forcedeletegod(self, ctx, arg1):
...
+            await ctx.send("Include both a name and new name!")
+            return
+        
+        if database.getGodName(args[1], ctx.guild.id):
+            await ctx.send("A God with that name already exists!")
+            return
+
+        if len(args[0]) > 16:
+            await ctx.send("Please choose a name that's not longer than 16 characters!")
+            return
```
Comment (HeroGamers) [inline cogs/AdminManager.py]: You accidentally switched the two args around here.

File: cogs/GodManager.py [hunk 11]
```diff
@@ -168,12 +168,19 @@ async def _setgender(self, ctx, arg1):
...
     @commands.check(utilchecks.isPriest)
     async def _rename(self, ctx, arg1):
         """Rename existing God."""
-        god = database.getBeliever(ctx.authos.id, ctx.guild.id).God
-        
+        if database.getGodName(arg1, ctx.guild.id):
+            await ctx.send("A God with that name already exists!")
+            return
+
+        if len(args[0]) > 16:
```
Comment (HeroGamers) [inline cogs/GodManager.py]: args doesn't exist here. You have only provided the arg1 argument.

**Review by HeroGamers — CHANGES_REQUESTED** [review]
Almost there, just a few things now!
You'd probably want to look into using a dedicated IDE, like PyCharm or Visual Studio Code with Pylint installed.

You're starting to get the hang of it. Lmk if you have any further questions on Discord or just comment here.

**Comment by mrSheploo** [comment]
Been very busy in a past month due to new job.
I hope I fixed what You've mentioned.
Also, yes, I'll deffinately try dedicated IDE

**Comment by HeroGamers** [comment]
Closing PR as it's stale.

**Pull request closed without merging**